tests/services/phase.service.spec.ts

```diff
@@ -0,0 +1,119 @@
+      phaseService = new PhaseService(null, githubService, null, userService, null);
+    });
+
+    describe('.fetchSessionData()', () => {
+      it('should fetch the correct settings file', () => {
+        githubService.fetchSettingsFile.and.returnValue(of(mockSettingsFile));
+        phaseService.fetchSessionData().subscribe((sessionData: SessionData) => {
+          expect(sessionData).toEqual(mockSessionData);
+        });
+      });
+    });
+
+    describe('.updateSessionParameters()', () => {
+      it('should update the currentPhase and repoName based on given sessionData', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+
+        expect(phaseService.currentPhase).toEqual(Phase.phaseBugReporting);
+        expect(phaseService.repoName).toEqual(mockSettingsFile[Phase.phaseBugReporting]);
+      });
+    });
+
+    describe('.attemptSessionAvailabilityFix()', () => {
+      it('should throw no errors given phaseBugReporting and student role', () => {
+        userService.currentUser = testStudent;
+        // refresh phaseService with new userService
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).not.toThrowError();
+      });
+
+      it('should throw an error given no openPhases', () => {
+        userService.currentUser = testStudent;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Current Phase\'s Repository has not been opened.')
+        );
+      });
+
+      it('should throw an error given non-student role and phaseBugReporting', () => {
+        userService.currentUser = testTutor;
+        phaseService = new PhaseService(null, githubService, null, userService, null);
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.updateSessionParameters(mockSessionData);
+        githubService.createRepository.and.callFake(() => {});
+        expect(() => phaseService.attemptSessionAvailabilityFix(mockSessionData)).toThrow(
+          new Error('Bug-Reporting Phase\'s repository initialisation is only available to Students.')
+        );
+      });
+    });
+
+    describe('.githubRepoPermissionLevel()', () => {
+      it('should return "repo" if phaseModeration is included in openPhases', () => {
+        githubService.storePhaseDetails.and.callFake(() => {});
+        phaseService.sessionData = {
+          ...mockSessionData,
+          openPhases: [ Phase.phaseModeration ]
```

Comment:
Minor suggestion but maybe can remove the spaces around the `[` and `]` to keep the `openPhases` array consistent with your initial definition of `mockSessionData` which doesn't have spaces?

Reply:
Alright! Yes I made an oopsie on my part!

Reply:
No probs!